Engine/src/Engine/ResourceManagement/ShaderNorms.h

```diff
@@ -2,6 +2,9 @@
 #ifndef SHADERNORMS_H
 #define SHADERNORMS_H
 
+#include <Engine/ResourceManagement/ShaderFillType.h>
+#include <Engine/ResourceManagement/ShaderGenerator.h>
+#include <Engine/ResourceManagement/AdvancedShaderDistributor.h>
```

Comment:
this and shader generator can be forward declared

Reply:
Done

Reply:
So this is still providing errors for the engine since the header file is dependent on certain components of the forward-declared classes i.e. ShaderFillType::FillType which doesn't get read in the forward declare.

Reply:
Actually, that's just shaderfilltype so I did it for the others

Reply:
Take a look at https://stackoverflow.com/questions/71416/forward-declaring-an-enum-in-c to declare enums forward.
If that still doesn't work then keep the include, but get rid of the forward declaration of any included type. Though you should be able to forward declare all 3 of these.